### PR TITLE
App insights client logger with example

### PR DESCRIPTION
**Pull request by seanimam**
### This PR creates two packages:
1. `@fluidframework/app-insights-logger"`provides a telemetry logger that will route typical fluid telemetry to Azure App Insights.
2. `@fluid-example/app-insights-logger` Aa simple fluid application complete with a UI view in react to test the fluid App Insights telemetry logger.

### Starting the example app: 

1. #### Configure the logger to send telemtry to your app insights instance
    -  within `ClientUtilities.ts`, update the function definition for `initializeTinyliciousClient` to initialize the the ApplicationInsights client with your instances configuration. In most cases, this is simply the most basic config containing your correct connection string.
2. #### run `pnpm run start:test-app` and navigate to http://localhost:8080/ in your web browser.

### Finding your metrics
-   Go to your app insights table and search the customEvents table to find all telemetry events sent by default.

## Breaking Changes
- None

### CODE REVIEW

File: examples/client-logger/app-insights-logger/.gitignore [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Generated by npm
```
Comment (Josmithr) [inline examples/client-logger/app-insights-logger/.gitignore:1]: Does this specify anything that the root .gitignore doesn't? If not, we may be able to remove it.

File: examples/client-logger/app-insights-logger/README.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# @fluid-example/app-insights-logger
```
Comment (Josmithr) [inline examples/client-logger/app-insights-logger/README.md:1]: I recommend checking out tools/api-markdown-documenter/README.md. It takes advantage of a number of our template macros for baking in various bits of useful information like our trademark info, contribution guidelines, etc. Some of our other sample app packages also use a macro for including app launching instructions that might be useful here.

File: examples/client-logger/app-insights-logger/jest.config.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+module.exports = {
```
Comment (Josmithr) [inline examples/client-logger/app-insights-logger/jest.config.js:6]: I would recommend mimicking the test pattern used in our other example apps - that way you can get (at least) a smoke test up and running for the app. `external-data` might be a good point of reference, though it does some extra server setup steps that won't be needed here.

File: packages/framework/client-logger/app-insights-logger/README.md [hunk 4]
```diff
@@ -0,0 +1,3 @@
+# @fluidframework/app-insights-logger
```
Comment (Josmithr) [inline packages/framework/client-logger/app-insights-logger/README.md:1]: Same comments as the other README, but this README also really needs usage documentation as well. E.g.,:
- What are the use cases for this? What does this solve for potential consumers?
- How does the user initialize it? Some code examples would be useful (We can also probably point to the sample app package as an example).
-

File: packages/framework/client-logger/app-insights-logger/api-report/app-insights-logger.api.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+
+import { ApplicationInsights } from '@microsoft/applicationinsights-web';
+import { ITelemetryBaseEvent } from '@fluidframework/common-definitions';
+import { ITelemetryBaseLogger } from '@fluidframework/common-definitions';
+
+// @public (undocumented)
```
Comment (Josmithr) [inline packages/framework/client-logger/app-insights-logger/api-report/app-insights-logger.api.md:11]: Let's be sure to document the undocumented items here.

File: packages/framework/client-logger/app-insights-logger/jest.config.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+module.exports = {
```
Comment (Josmithr) [inline packages/framework/client-logger/app-insights-logger/jest.config.js:6]: Do we need jest in this package? Can we get away with just using `mocha` w/ unit tests? We should also add some simple unit tests if we can (packages with no tests always make me nervous).

File: packages/framework/client-logger/app-insights-logger/src/fluidAppInsightsLogger.ts [hunk 7]
```diff
@@ -0,0 +1,26 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ApplicationInsights } from "@microsoft/applicationinsights-web";
+import { ITelemetryBaseEvent, ITelemetryBaseLogger } from "@fluidframework/common-definitions";
+
+export class FluidAppInsightsLogger implements ITelemetryBaseLogger {
+	protected readonly baseLoggingClient: ApplicationInsights;
```
Comment (Josmithr) [inline packages/framework/client-logger/app-insights-logger/src/fluidAppInsightsLogger.ts:10]: Are consumers of this type expected to sub-class it? If not, can we make this private? We can also mark the class as `@sealed` if this is the case (we can always un-seal it later if we wish, but this protects us from having to support certain classes of forwards/backwards compatibility).

File: packages/framework/client-logger/app-insights-logger/src/index.ts [hunk 8]
```diff
@@ -0,0 +1,12 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/**
+ * This package provides a telemetry logger that will route typical Fluid telemetry to Azure App Insights.
```
Comment (Josmithr) [inline packages/framework/client-logger/app-insights-logger/src/index.ts:7]: Nit: We can use an `{@link }` comment to point readers to documentation on Azure App Insights. E.g.,
```typescript
* ... {@link <url> | Azure App Insights}.
```

**Review by Josmithr — COMMENTED** [review]
I left a few comments, mostly around documentation and testing. Direction looks good to me though!

File: packages/framework/client-logger/app-insights-logger/package.json [hunk 9]
```diff
@@ -0,0 +1,85 @@
+{
+	"name": "@fluidframework/app-insights-logger",
```
Comment (Josmithr) [inline packages/framework/client-logger/app-insights-logger/package.json:2]: Nit: It might be a good idea to start with this package being set to `private`. We can do a separate PR to make it public once we are confident that everything is working as we want. As soon as we mark it as public, we are making compatibility guarantees.

Comment (seanimam) [reply packages/framework/client-logger/app-insights-logger/package.json:2]: Okay, I got a policy violation for it, apparently they have to be public in the framework package. There might be a way to add an exclusion

File: packages/framework/client-logger/app-insights-logger/src/index.ts (hunk 8 above)
Comment (seanimam) [reply packages/framework/client-logger/app-insights-logger/src/index.ts:7]: Done

File: packages/framework/client-logger/app-insights-logger/src/fluidAppInsightsLogger.ts (hunk 7 above)
Comment (seanimam) [reply packages/framework/client-logger/app-insights-logger/src/fluidAppInsightsLogger.ts:18]: Yeah there was an expectation that consumers would subclass it.

File: packages/framework/client-logger/app-insights-logger/jest.config.js (hunk 6 above)
Comment (seanimam) [reply packages/framework/client-logger/app-insights-logger/jest.config.js]: Switched from mocha to jest and added unit tests

File: packages/framework/client-logger/app-insights-logger/api-report/app-insights-logger.api.md (hunk 5 above)
Comment (seanimam) [reply packages/framework/client-logger/app-insights-logger/api-report/app-insights-logger.api.md]: Everything should be documented now

File: packages/framework/client-logger/app-insights-logger/README.md (hunk 4 above)
Comment (seanimam) [reply packages/framework/client-logger/app-insights-logger/README.md:1]: Added a readme.

File: examples/client-logger/app-insights-logger/jest.config.js (hunk 3 above)
Comment (seanimam) [reply examples/client-logger/app-insights-logger/jest.config.js:6]: I spent all day trying to get an integration test to work and failed to so.. But I build a much simpler test and confirms it renders.

File: packages/framework/client-logger/app-insights-logger/package.json (hunk 9 above)
Comment (alexvy86) [reply packages/framework/client-logger/app-insights-logger/package.json:2]: An alternative would be to put it in `@fluid-experimental` first. I remember seeing some discussion about that being an ok starting path to publish new packages, and move them to `@fluidframework` once it's proven itself.

File: packages/framework/client-logger/app-insights-logger/src/fluidAppInsightsLogger.ts (hunk 7 above)
Comment (alexvy86) [reply packages/framework/client-logger/app-insights-logger/src/fluidAppInsightsLogger.ts:18]: I still think there should not be such an expectation ([original argument](https://github.com/microsoft/FluidFramework/pull/14848#discussion_r1153449263) still unchanged). Ever more so now that we're asking the user to pass in an instance of the AppInsights client; they can keep a reference to it and update if they want, so exposing the underlying client here is redundant. I suggest we remove this and mark the class `@sealed` as @Josmithr said.

File: packages/framework/client-logger/app-insights-logger/src/fluidAppInsightsLogger.ts [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+ */
+export class FluidAppInsightsLogger implements ITelemetryBaseLogger {
+	/**
+	 * The Azure ApplicationInsights client utilized by this logger
+	 */
+	protected readonly baseLoggingClient: ApplicationInsights;
+
+	public constructor(client: ApplicationInsights) {
+		this.baseLoggingClient = client;
+		this.baseLoggingClient.loadAppInsights();
```
Comment (alexvy86) [inline packages/framework/client-logger/app-insights-logger/src/fluidAppInsightsLogger.ts:22]: [Still think](https://github.com/microsoft/FluidFramework/pull/14848#discussion_r1153506090) that we should tell consumers that they need to give us a fully initialized app insights client. If there's a way to check if `loadAppInsights()` was already called on it, I'd even suggest we check and throw an error if it hasn't been (i.e. the client is not in the state we expect it).

I tried to look at the code; it would seem calling `loadAppInsights()` twice won't throw or anything, but I'd still prefer that we completely get out of the business of setting up the client, since there's nothing we need configured in a particular way.

File: packages/framework/client-logger/app-insights-logger/package.json [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+		"eslint-plugin-react-hooks": "~4.6.0",
+		"mocha": "^10.2.0",
+		"prettier": "~2.6.2",
+		"rimraf": "^4.4.0",
+		"sinon": "^7.4.2",
+		"tinylicious": "0.7.2",
+		"ts-loader": "^9.3.0",
+		"tslib": "^1.10.0",
+		"typescript": "~4.5.5",
+		"webpack": "^5.76.2"
```
Comment (alexvy86) [inline packages/framework/client-logger/app-insights-logger/package.json:63]: `tinylicious` and `webpack` can probably go away, they don't seem used in this package anymore. A bit less familiar with `ts-loader` and `tslib`, but I suspect also unnecessary? `ts-loader` seems relevant to the webpack configuration that moved to the example package.

File: packages/framework/client-logger/app-insights-logger/package.json [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+					"dist/main.bundle.js"
+				],
+				"library": "main"
+			}
+		}
+	},
+	"jest-junit": {
+		"outputDirectory": "nyc",
+		"outputName": "jest-junit-report.xml"
+	},
```
Comment (alexvy86) [inline packages/framework/client-logger/app-insights-logger/package.json:78]: Since we moved away from jest this should go away, right?

File: examples/client-logger/app-insights-logger/package.json [hunk 13]
```diff
@@ -0,0 +1,122 @@
...
+		"@fluidframework/fluid-static": ">=2.0.0-internal.4.1.0 <2.0.0-internal.4.2.0",
+		"@fluidframework/map": ">=2.0.0-internal.4.1.0 <2.0.0-internal.4.2.0",
+		"@fluidframework/protocol-definitions": "^1.1.0",
+		"@fluidframework/sequence": ">=2.0.0-internal.4.1.0 <2.0.0-internal.4.2.0",
+		"@fluidframework/shared-object-base": ">=2.0.0-internal.4.1.0 <2.0.0-internal.4.2.0",
+		"@fluidframework/tinylicious-client": ">=2.0.0-internal.4.1.0 <2.0.0-internal.4.2.0",
+		"@microsoft/applicationinsights-web": "^2.8.11",
+		"react": "^17.0.1",
+		"react-dom": "^17.0.1"
+	},
```
Comment (alexvy86) [inline examples/client-logger/app-insights-logger/package.json:64]: Let's clean this up, some of these aren't direct dependencies of the package so shouldn't need to be here.

Same for devDependencies. `@fluid-example/collaborative-textarea`, for example, isn't used AFAICT.

File: packages/framework/client-logger/app-insights-logger/src/fluidAppInsightsLogger.ts (hunk 10 above)
Comment (seanimam) [reply packages/framework/client-logger/app-insights-logger/src/fluidAppInsightsLogger.ts:22]: I tested this and calling it twice seemed to be fine but there could perhaps be some kind of a side effect so I can remove this and update the documentation accordingly.

File: packages/framework/client-logger/app-insights-logger/README.md [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+	logger.send({category: "mockEvent",	eventName: "mockEventName"});
+```
+
+## Viewing Logs in App Insights:
+
+From the Azure web portal, navigate to your app insights instance. Now, go to the "Logs" for your instance, this should be an option within the left side panel. Finally, from this page, you can query for telemetry events, which will be stored in the customEvents table. As an example, you can issue this simple query to get recent telemetry events sent to the customEvents table:
+
+```
+customEvents
+```
```
[the remark this answers is not in the document]
Comment (seanimam) [reply packages/framework/client-logger/app-insights-logger/README.md:42]: That is the query lol.. But I can add some more to it since its so basic

File: examples/client-logger/app-insights-logger/.gitignore (hunk 1 above)
Comment (seanimam) [reply examples/client-logger/app-insights-logger/.gitignore]: Removed this file

File: packages/framework/client-logger/app-insights-logger/README.md
[the remark this answers is not in the document]
Comment (seanimam) [reply packages/framework/client-logger/app-insights-logger/README.md:1]: Okay, added this callout.

File: packages/framework/client-logger/app-insights-logger/README.md [hunk 15]
```diff
@@ -0,0 +1,42 @@
...
+You create an App Insights Instance and then configure your applications to send data to your instance using either Azure provided SDK’s or REST API’s.
+This could be general machine related health being automatically reported to the instance when you install a logging program on your service’s machines or custom metrics that you manually configure your applications to send.
+In our case, we are sending custom metrics.[ Learn more about Azure app Insights with their docs](https://learn.microsoft.com/en-us/azure/azure-monitor/app/app-insights-overview?tabs=net)
+
+## Use case:
+
+The primary use case of sending telemetry to Azure App Insights allows users to analyze and store telemetry logs without having to manually setup complex infrastructure.
+Once in App Insights users can leverage Azure's log exploring tools to analyze logs.
+
+## Usage:
```
[the remark this answers is not in the document]
Comment (seanimam) [reply packages/framework/client-logger/app-insights-logger/README.md:17]: Yep, added this step in the example under usage

File: packages/framework/client-logger/app-insights-logger/README.md [hunk 16]
```diff
@@ -0,0 +1,42 @@
...
+	const appInsightsClient = new ApplicationInsights({
+		config: {
+			connectionString:
+				"InstrumentationKey=abcdefgh-ijkl-mnop-qrst-uvwxyz6ffd9c;IngestionEndpoint=https://westus2-2.in.applicationinsights.azure.com/;LiveEndpoint=https://westus2.livediagnostics.monitor.azure.com/",
+		},
+	});
+
+	const logger = new FluidAppInsightsLogger(appInsightsClient);
+
+	logger.send({category: "mockEvent",	eventName: "mockEventName"});
```
[the remark this answers is not in the document]
Comment (seanimam) [reply packages/framework/client-logger/app-insights-logger/README.md:37]: Updated the readme to include this.

File: packages/framework/client-logger/app-insights-logger/src/fluidAppInsightsLogger.ts (hunk 7 above)
Comment (seanimam) [reply packages/framework/client-logger/app-insights-logger/src/fluidAppInsightsLogger.ts:20]: I've added the @sealed mark.

File: packages/framework/client-logger/app-insights-logger/package.json (hunk 12 above)
Comment (seanimam) [reply packages/framework/client-logger/app-insights-logger/package.json]: Right. I have removed this file

File: examples/client-logger/app-insights-logger/package.json (hunk 13 above)
Comment (seanimam) [reply examples/client-logger/app-insights-logger/package.json]: I've cleaned up the dependency section from all the unused ones I could see. Should be good to go now.

File: packages/framework/client-logger/app-insights-logger/package.json (hunk 11 above)
Comment (seanimam) [reply packages/framework/client-logger/app-insights-logger/package.json]: `tslib` is required for app insights, but I removed `tinylicious` `webpack` & `ts-loader`

File: packages/framework/client-logger/app-insights-logger/README.md (hunk 14 above)
[the remark this answers is not in the document]
Comment (seanimam) [reply packages/framework/client-logger/app-insights-logger/README.md]: Updated this section with better query examples

File: packages/framework/client-logger/app-insights-logger/README.md [hunk 17]
```diff
@@ -0,0 +1,42 @@
+# @fluidframework/app-insights-logger
+
+## This `app-insights-logger` package provides a Fluid telemetry logger that will route typical Fluid telemetry to Azure App Insights using the `ApplicationInsights.trackEvent` API proided by the `@microsoft/applicationinsights-web` package.
+
+## What is Application Insights?
+
+At a high level, App Insights is an Azure cloud service for aggregating, visualizing, analyzing, and alerting on metrics related to a given “service” or “application”.
+You create an App Insights Instance and then configure your applications to send data to your instance using either Azure provided SDK’s or REST API’s.
+This could be general machine related health being automatically reported to the instance when you install a logging program on your service’s machines or custom metrics that you manually configure your applications to send.
+In our case, we are sending custom metrics.[ Learn more about Azure app Insights with their docs](https://learn.microsoft.com/en-us/azure/azure-monitor/app/app-insights-overview?tabs=net)
```
[the remark this answers is not in the document]
Comment (seanimam) [reply packages/framework/client-logger/app-insights-logger/README.md]: Fixed.

File: packages/framework/client-logger/app-insights-logger/tsconfig.json [hunk 18]
```diff
@@ -0,0 +1,12 @@
+{
+	"extends": "@fluidframework/build-common/ts-common-config.json",
+	"compilerOptions": {
+		"rootDir": "src",
+		"outDir": "dist",
+		"composite": true,
+		"target": "es6",
+		"jsx": "react",
```
[the remark this answers is not in the document]
Comment (seanimam) [reply packages/framework/client-logger/app-insights-logger/tsconfig.json]: I have removed these two lines.

File: packages/framework/client-logger/app-insights-logger/README.md [hunk 19]
```diff
@@ -0,0 +1,64 @@
...
+```json
+	const appInsightsClient = new ApplicationInsights({
+		config: {
+			connectionString:
+				"InstrumentationKey=abcdefgh-ijkl-mnop-qrst-uvwxyz6ffd9c;IngestionEndpoint=https://westus2-2.in.applicationinsights.azure.com/;LiveEndpoint=https://westus2.livediagnostics.monitor.azure.com/",
+		},
+	});
+
+   // Initializes the App Insights client. Without this, logs will not be sent to Azure.
+	appInsightsClient.loadAppInsights()
```
Comment (alexvy86) [inline packages/framework/client-logger/app-insights-logger/README.md:32]: Indentation nit
```suggestion
    // Initializes the App Insights client. Without this, logs will not be sent to Azure.
	appInsightsClient.loadAppInsights()
```

File: examples/client-logger/app-insights-logger/src/test/components/App.test.tsx [hunk 20]
```diff
@@ -0,0 +1,17 @@
...
+import { render, screen } from "@testing-library/react";
+import React from "react";
+import { App } from "../../components";
+
+describe("App Insights Example App UI test", () => {
+	it("App renders", async (): Promise<void> => {
+		render(<App />);
+		await screen.findByText("Loading Shared container...");
+	});
+});
```
Comment (alexvy86) [inline examples/client-logger/app-insights-logger/src/test/components/App.test.tsx:17]: It seems like `@testing-library/jest-dom` gives us DOM-related extensions for assertions, like `expect(<some-dom-node>).toBeDisabled()`... let's either use it here if relevant or remove the import (and the dependency altogether if we don't actually use it).

File: examples/client-logger/app-insights-logger/src/components/ClientUtilities.ts [hunk 21]
```diff
@@ -0,0 +1,151 @@
...
+function initializeTinyliciousClient(): TinyliciousClient {
+	const appInsightsClient = new ApplicationInsights({
+		config: {
+			connectionString:
+				// Edit this with your app insights instance connection string (this is an example string)
+				"InstrumentationKey=abcdefgh-ijkl-mnop-qrst-uvwxyz6ffd9c;IngestionEndpoint=https://westus2-2.in.applicationinsights.azure.com/;LiveEndpoint=https://westus2.livediagnostics.monitor.azure.com/",
+		},
+	});
+
+	// appInsightsClient.loadAppInsights();
```
Comment (alexvy86) [inline examples/client-logger/app-insights-logger/src/components/ClientUtilities.ts:50]: Should this be uncommented?

**Review by alexvy86 — COMMENTED** [review]
We're almost there, mostly some remaining cleanup.

File: examples/client-logger/app-insights-logger/README.md (hunk 2 above)
Comment (Josmithr) [reply examples/client-logger/app-insights-logger/README.md:1]: Note: in order to generate the contents, you'll have to run a build from the docs directory (for now).

File: packages/framework/client-logger/app-insights-logger/README.md [hunk 22]
```diff
@@ -0,0 +1,64 @@
+# @fluid-internal/app-insights-logger
+
+## This `app-insights-logger` package provides a Fluid telemetry logger that will route Fluid telemetry to Azure App Insights using the `ApplicationInsights.trackEvent` API provided by the `@microsoft/applicationinsights-web` package. The logger is intended for use by browser based web applications, not NodeJS applications [as stated by the App Insights Web SDK.](https://learn.microsoft.com/en-us/azure/azure-monitor/app/app-insights-overview?tabs=net)
```
Comment (Josmithr) [inline packages/framework/client-logger/app-insights-logger/README.md:3]: Nit: I don't think this wants to be a heading.
```suggestion
This `app-insights-logger` package provides a Fluid telemetry logger that will route Fluid telemetry to Azure App Insights using the `ApplicationInsights.trackEvent` API provided by the `@microsoft/applicationinsights-web` package. The logger is intended for use by browser based web applications, not NodeJS applications [as stated by the App Insights Web SDK.](https://learn.microsoft.com/en-us/azure/azure-monitor/app/app-insights-overview?tabs=net)
```

File: packages/framework/client-logger/app-insights-logger/README.md [hunk 23]
```diff
@@ -0,0 +1,64 @@
...
+
+## What is Application Insights?
+
+At a high level, App Insights is an Azure cloud service for aggregating, visualizing, analyzing, and alerting on metrics related to a given “service” or “application”.
+You create an App Insights Instance and then configure your applications to send data to your instance using either Azure provided SDK’s or REST API’s.
```
Comment (Josmithr) [inline packages/framework/client-logger/app-insights-logger/README.md:8]: Nit:
```suggestion
You create an App Insights Instance and then configure your applications to send data to your instance using either Azure provided SDK’s or REST APIs.
```

File: packages/framework/client-logger/app-insights-logger/README.md [hunk 24]
```diff
@@ -0,0 +1,64 @@
...
+This could be general machine related health being automatically reported to the instance when you install a logging program on your service’s machines or custom metrics that you manually configure your applications to send. Keep in mind this logger is intended for use with browser based web applications, not pure nodeJS.
+In our case, we are sending custom metrics. [Learn more about Azure App Insights with their docs](https://learn.microsoft.com/en-us/azure/azure-monitor/app/app-insights-overview?tabs=net)
+
+## Use case:
```
Comment (Josmithr) [inline packages/framework/client-logger/app-insights-logger/README.md:12]: Nit: Since this is a heading, it probably doesn't need a `:`
```suggestion
## Use case
```

Comment (Josmithr) [reply packages/framework/client-logger/app-insights-logger/README.md:12]: (same with some of the other headings)

File: packages/framework/client-logger/app-insights-logger/README.md [hunk 25]
```diff
@@ -0,0 +1,64 @@
...
+The primary use case of sending telemetry to Azure App Insights allows users to analyze and store telemetry logs without having to manually setup complex infrastructure.
+Once in App Insights users can leverage Azure's log exploring tools to analyze logs.
+
+## Usage:
+
+In order to use this logger, users will first have to initialize an `ApplicationInsights` client from the `@microsoft/applicationinsights-web` package. In most cases, initializing the `ApplicationInsights` client will be just providing your App Insight instances connection string AND calling `.loadAppInsights()` on your client. [Learn more about the App Insights SDK](https://github.com/microsoft/ApplicationInsights-JS#before-getting-started)
+
+Here is an example usage:
+
+```json
```
Comment (Josmithr) [inline packages/framework/client-logger/app-insights-logger/README.md:23]: This should be specifying `typescript` or `javascript`, right?

File: examples/client-logger/app-insights-logger/README.md [hunk 26]
```diff
@@ -0,0 +1,70 @@
...
+
+## Overview
+
+This package provides a simple Fluid application complete with a UI view in [React](https://react.dev/) to test the Fluid App Insights telemetry logger that will route typical Fluid telemetry to configured Azure App Insights.
+
+<!-- AUTO-GENERATED-CONTENT:START (README_INSTALLATION_SECTION:includeHeading=TRUE&devDependency=TRUE) -->
```
Comment (Josmithr) [inline examples/client-logger/app-insights-logger/README.md:7]: I'm assuming consumers will generally use this as a non-dev dependency. So I would probably update to the following (and re-run generation):
```suggestion
<!-- AUTO-GENERATED-CONTENT:START (README_INSTALLATION_SECTION:includeHeading=TRUE&devDependency=FALSE) -->
```

File: examples/client-logger/app-insights-logger/README.md [hunk 27]
```diff
@@ -0,0 +1,70 @@
...
+```
+
+## Starting the test
+
+-   run `pnpm run start:test-app` and navigate to http://localhost:8080/ in your web browser.
+
+## Generating telemetry events
+
+-   There will be telemetry events that flow automatically when you start the test app. In addition to these events, you can control creating telemetry events yourself by interacting with UI app, incremeting/decrementing the shared counter and editing the shared string provided in this example
+
```
Comment (Josmithr) [inline examples/client-logger/app-insights-logger/README.md:53]: Nit: It would also probably be nice to add the "API Documentation", "Help", and "Contribution Guidelines" sections here as well.

File: examples/client-logger/app-insights-logger/README.md (hunk 26 above)
Comment (Josmithr) [inline examples/client-logger/app-insights-logger/README.md:7]: Since this is an example app project, we don't presumably don't expect external consumers to install this. We should probably remove this section from this README.

Comment (Josmithr) [reply examples/client-logger/app-insights-logger/README.md:7]: Some of the other sample apps use one of our pre-canned pragmas for instructions on how to run the app, though (with an optional argument for whether or not running requires tinylicious). I would probably use that instead.

File: examples/client-logger/app-insights-logger/src/components/App.tsx [hunk 28]
```diff
@@ -0,0 +1,261 @@
...
+
+/**
+ * Root application component.
+ * Initializes the Fluid Container and displays app view once it is ready.
+ */
+export function App(): React.ReactElement {
+	// Load the collaborative SharedString object
+	// const { privateContainer, sharedContainer } = useContainerInfo();
+
+	const [sharedContainerInfo, setSharedContainerInfo] = React.useState<
```
Comment (Josmithr) [inline examples/client-logger/app-insights-logger/src/components/App.tsx:81]: Nit: Since you only have a single container, you could probably just call this `container` and `setContainer`.

File: examples/client-logger/app-insights-logger/webpack.config.js [hunk 29]
```diff
@@ -0,0 +1,64 @@
...
+			module: {
+				rules: [
+					{
+						test: /\.tsx?$/,
+						loader: require.resolve("ts-loader"),
+					},
+					{
+						test: /\.css$/,
+						use: [
+							require.resolve("css-loader"), // translates CSS into CommonJS
```
Comment (Josmithr) [inline examples/client-logger/app-insights-logger/webpack.config.js:35]: Are these used?

File: examples/client-logger/app-insights-logger/webpack.config.js [hunk 30]
```diff
@@ -0,0 +1,64 @@
...
+			output: {
+				filename: "[name].bundle.js",
+				path: path.resolve(__dirname, "dist"),
+				library: "[name]",
+				devtoolNamespace: "fluid-tools/client-debugger-view",
+				libraryTarget: "umd",
+			},
+			plugins: [
+				new webpack.ProvidePlugin({
+					process: require.resolve("process/browser"),
```
Comment (Josmithr) [inline examples/client-logger/app-insights-logger/webpack.config.js:50]: @alexvy86 has a PR up for client-devtools-view removing this. I suspect we don't need this here either?

File: packages/framework/client-logger/app-insights-logger/.eslintrc.js [hunk 31]
```diff
@@ -0,0 +1,11 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+module.exports = {
+	extends: [require.resolve("@fluidframework/eslint-config-fluid/minimal"), "prettier"],
```
Comment (Josmithr) [inline packages/framework/client-logger/app-insights-logger/.eslintrc.js:7]: We shouldn't be using the minimal config in any new packages. I would recommend using the strict variant if possible. Otherwise recommended.

File: examples/client-logger/app-insights-logger/.eslintrc.js [hunk 32]
```diff
@@ -0,0 +1,11 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+module.exports = {
+	extends: [require.resolve("@fluidframework/eslint-config-fluid/minimal"), "prettier"],
```
Comment (Josmithr) [inline examples/client-logger/app-insights-logger/.eslintrc.js:7]: We shouldn't be using the minimal config in any new packages. `strict` is always my recommendation, though `recommended` is fine for test apps.

File: packages/framework/client-logger/app-insights-logger/README.md [hunk 33]
```diff
@@ -0,0 +1,110 @@
+# @fluid-internal/app-insights-logger
+
+## Overview
+
+This `app-insights-logger` package provides a Fluid telemetry logger that will route Fluid telemetry to Azure App Insights using the `ApplicationInsights.trackEvent` API provided by the `@microsoft/applicationinsights-web` package. The logger is intended for use by browser based web applications, not NodeJS applications [as stated by the App Insights Web SDK.](https://learn.microsoft.com/en-us/azure/azure-monitor/app/app-insights-overview?tabs=net)
```
Comment (Josmithr) [inline packages/framework/client-logger/app-insights-logger/README.md:5]: Nit: 
```suggestion
This `app-insights-logger` package provides a Fluid telemetry logger that will route Fluid telemetry to Azure App Insights using the `ApplicationInsights.trackEvent` API provided by the [@microsoft/applicationinsights-web](https://www.npmjs.com/package/@microsoft/applicationinsights-web) package. The logger is intended for use by browser based web applications, not NodeJS applications [as stated by the App Insights Web SDK.](https://learn.microsoft.com/en-us/azure/azure-monitor/app/app-insights-overview?tabs=net)
```

File: packages/framework/client-logger/app-insights-logger/README.md [hunk 34]
```diff
@@ -0,0 +1,110 @@
...
+
+## What is Application Insights?
+
+At a high level, App Insights is an Azure cloud service for aggregating, visualizing, analyzing, and alerting on metrics related to a given “service” or “application”.
+You create an App Insights Instance and then configure your applications to send data to your instance using either Azure provided SDK’s or REST APIs.
+This could be general machine related health being automatically reported to the instance when you install a logging program on your service’s machines or custom metrics that you manually configure your applications to send. Keep in mind this logger is intended for use with browser based web applications, not pure nodeJS.
+In our case, we are sending custom metrics. [Learn more about Azure App Insights with their docs](https://learn.microsoft.com/en-us/azure/azure-monitor/app/app-insights-overview?tabs=net)
+
+<!-- AUTO-GENERATED-CONTENT:START (README_INSTALLATION_SECTION:includeHeading=TRUE&devDependency=TRUE) -->
```
Comment (Josmithr) [inline packages/framework/client-logger/app-insights-logger/README.md:14]: I'm assuming we expect consumers of this library to use it as a non-dev dependency. I would recommend making the following change and rerunning the gen step:
```suggestion
<!-- AUTO-GENERATED-CONTENT:START (README_INSTALLATION_SECTION:includeHeading=TRUE&devDependency=FALSE) -->
```

**Review by Josmithr — COMMENTED** [review]
I left a few more comments. These changes look good to me, except that I think we need to migrate away from the "minimal" eslint configs in both new packages. The minimal config should be considered deprecated - we don't want any new usages of it. And for the library package especially, we should be using the `strict` config.

More details on the differences and when/where to use them: https://github.com/microsoft/FluidFramework/tree/main/common/build/eslint-config-fluid#configurations

**Review by Josmithr — APPROVED** [review]
Looks good to me!

**Review by alexvy86 — APPROVED** [review]
🚀
